internal/raft/message/convert.go

```diff
@@ -0,0 +1,971 @@
+	return msgCmdLimit
+}
+
+// ConvertCommandToMessageInsertOr converts command.InsertOr to a message.InsertOr
+func ConvertCommandToMessageInsertOr(cmd command.InsertOr) InsertOr {
+	switch cmd {
+	case 0:
+		return InsertOr_InsertOrUnknown
+	case 1:
+		return InsertOr_InsertOrReplace
+	case 2:
+		return InsertOr_InsertOrRollback
+	case 3:
+		return InsertOr_InsertOrAbort
+	case 4:
+		return InsertOr_InsertOrFail
+	case 5:
+		return InsertOr_InsertOrIgnore
+	}
+	return -1
+}
+
+// ConvertCommandToMessageInsert converts a Command type to a Command_Insert type.
+func ConvertCommandToMessageInsert(cmd command.Command) *Command_Insert {
+	msgCmdInsert := &Command_Insert{}
+	msgCmdInsert.InsertOr = ConvertCommandToMessageInsertOr(cmd.(*command.Insert).InsertOr)
+	msgCmdInsert.Table = ConvertCommandToMessageTable(cmd.(*command.Insert).Table)
+	msgCmdInsert.Cols = ConvertCommandToMessageCols(cmd.(*command.Insert).Cols)
+	msgCmdInsert.DefaultValues = cmd.(*command.Insert).DefaultValues
+	msgCmdInsert.Input = ConvertCommandToMessageList(cmd.(*command.Insert).Input)
+	return msgCmdInsert
+}
+
+// ConvertMessageToCommand converts a message.Command to a command.Command
+func ConvertMessageToCommand(msg Message) command.Command {
+	switch m := msg.(type) {
+	case *Command_Scan:
+		return ConvertMessageToCommandScan(m)
+	case *Command_Select:
+		return ConvertMessageToCommandSelect(m)
+	case *Command_Project:
+		return ConvertMessageToCommandProject(m)
+	case *Command_Delete:
+		return ConvertMessageToCommandDelete(m)
+	case *CommandDrop:
+		switch m.Target {
+		case 0:
+			return ConvertMessageToCommandDropTable(m)
+		case 1:
+			return ConvertMessageToCommandDropView(m)
+		case 2:
+			return ConvertMessageToCommandDropIndex(m)
+		case 3:
+			return ConvertMessageToCommandDropTrigger(m)
+		}
+	case *Command_Update:
+		return ConvertMessageToCommandUpdate(m)
+	case *Command_Join:
+		return ConvertMessageToCommandJoin(m)
+	case *Command_Limit:
+		return ConvertMessageToCommandLimit(m)
+	case *Command_Insert:
+		return ConvertMessageToCommandInsert(m)
+	}
+	return nil
+}
+
+// ConvertMessageToCommandTable converts a message.SimpleTable to a command.Table
+func ConvertMessageToCommandTable(msg *SimpleTable) command.Table {
+	cmdTable := &command.SimpleTable{}
+	cmdTable.Schema = msg.Schema
+	cmdTable.Table = msg.Table
+	cmdTable.Alias = msg.Alias
+	cmdTable.Indexed = msg.Indexed
+	cmdTable.Index = msg.Index
+	return cmdTable
+}
+
+// ConvertMessageToCommandScan converts a message.Command_Scan to a command.Scan
+func ConvertMessageToCommandScan(msg *Command_Scan) *command.Scan {
+	cmdScan := &command.Scan{}
+	cmdScan.Table = ConvertMessageToCommandTable(msg.Table)
+	return cmdScan
+}
+
+// ConvertMessageToCommandLiteralExpr converts a message.Expr to a command.LiteralExpr
+func ConvertMessageToCommandLiteralExpr(msg *Expr) *command.LiteralExpr {
+	literalExpr := &command.LiteralExpr{}
+	literalExpr.Value = msg.GetLiteral().GetValue()
```

Comment:
Why are the getters neccessary?

Reply:
Getters were generated by protobuf, I just used them as they were there.